File: job-watchdog-starter/src/main/java/com/joshlong/watchdog/HeartbeatEvent.java
package com.joshlong.watchdog;

import org.springframework.context.ApplicationEvent;

/**
 * Broadcast to signal that the service is still alive.
 */
public class HeartbeatEvent extends ApplicationEvent {

    public HeartbeatEvent() {
        super(Watchdog.class.getName());
    }
}

File: src/main/kotlin/com/joshlong/jobs/watchdog/HeartbeatEvent.kt
package com.joshlong.jobs.watchdog

import org.springframework.context.ApplicationEvent


/**
 * Signals that a given JVM process should be maintained. This ultimately calls
 * {@link com.joshlong.twitter.InactivityMonitor#watch()} which has a daemon process running.
 *
 * @author <a href="mailto:josh@joshlong.com">Josh Long</a>
 */
class HeartbeatEvent() : ApplicationEvent(null)